src/compile/mark/area.ts

```diff
@@ -19,65 +19,134 @@ export namespace area {
 
     const stack = model.stack();
     const xFieldDef = model.encoding().x;
+    const x2FieldDef = model.encoding().x2;
     // x
     if (stack && X === stack.fieldChannel) { // Stacked Measure
       p.x = {
         scale: model.scaleName(X),
         field: model.field(X, { suffix: '_start' })
       };
-    } else if (isMeasure(xFieldDef)) { // Measure
-      p.x = { scale: model.scaleName(X), field: model.field(X) };
-    } else if (isDimension(xFieldDef)) {
-      p.x = {
-        scale: model.scaleName(X),
-        field: model.field(X, { binSuffix: '_mid' })
-      };
-    }
-
-    // x2
-    if (orient === 'horizontal') {
-      if (stack && X === stack.fieldChannel) {
+      if (orient === 'horizontal') {
         p.x2 = {
           scale: model.scaleName(X),
           field: model.field(X, { suffix: '_end' })
         };
+      }
+    } else if (isMeasure(xFieldDef) || isMeasure(x2FieldDef)) { // Measure
```

Comment:
Mostly the reason was for consistency with `rule` and `bar` where we had a similar logic. In the code before this we were doing `x` `x2` `y` y2` all separately and had independent `if` clauses for each of them. However, `x` & `x2` or `y` & `y2` are actually dependent on each other in some cases where we want to implicitly handle a incorrect `spec`. While reading through this code, I also discovered another problem in #1382 for `rule` and `bar` mark.